packages/sui-types/src/lib.rs

```diff
@@ -1,13 +1,20 @@
 use std::str::FromStr;
 
-use error_stack::{report, Report, Result, ResultExt};
+use error_stack::{ensure, report, Report, Result, ResultExt};
 use serde::{Deserialize, Serialize};
-
-use crate::error::Error;
+use thiserror::Error;
 
 const ADDRESS_PREFIX: &str = "0x";
 const SUI_ADDRESS_LENGTH: usize = 32;
 
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid address bytes: {:?}", .0)]
+    InvalidAddressBytes(Vec<u8>),
+    #[error("invalid address hex: {0}")]
+    InvalidAddressHex(String),
```

Comment:
```suggestion
    InvalidSuiAddress(String),
```
since this error is used for a few checks (lowercase + hex encoding + prefix stripping)

Reply:
it's already in the sui crate, no need to add sui to the variant name

Reply:
Sure, I'd still change it to `InvalidAddress` and not just `InvalidAddressHex` since some checks aren't about hex encoding.